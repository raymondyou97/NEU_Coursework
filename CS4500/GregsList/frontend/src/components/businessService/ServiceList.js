import React from 'react';
import PropTypes from 'prop-types';

import { ListGroup } from 'react-bootstrap';

const ServiceList = ({
  services,
  selectedService,
  setSelectedService,
  setSelectedServices,
}) => {
  if (services) {
    return (
      <ListGroup>
        {services.map(({ id, title }) => {
          const isSelected = selectedService === id;
          return (
            <ListGroup.Item
              key={id}
              active={isSelected}
              onClick={() => {
                setSelectedService(id === selectedService ? undefined : id);
                setSelectedServices(prevServices => {
                  if (prevServices.findIndex(ss => ss.id === id) === -1) {
                    return [...prevServices, { id, title }];
                  } else {
                    return prevServices;
                  }
                });
              }}
            >
              {title} âœ“
            </ListGroup.Item>
          );
        })}
      </ListGroup>
    );
  }

  return null;
};

ServiceList.propTypes = {
  services: PropTypes.arrayOf(PropTypes.object.isRequired),
  selectedService: PropTypes.number,
  setSelectedService: PropTypes.func.isRequired,
  setSelectedServices: PropTypes.func.isRequired,
};

export default ServiceList;
