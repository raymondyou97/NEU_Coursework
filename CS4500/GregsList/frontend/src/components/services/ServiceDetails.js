import React from 'react';
import ServiceService from '../../services/ServiceService';

import '../../styles/services/serviceDetails.css';

class ServiceDetails extends React.Component {
  constructor(props) {
    super(props);

    this.serviceService = ServiceService.getInstance();
    this.state = {
      services: [],
      title: '',
      id: this.props.match.params.id,
    };
  }

  componentDidMount() {
    this.serviceService.findAllServices().then(services => {
      this.props.history.push('/admin/services/' + services[0].id);
      this.setState({
        services: services,
      });
    });
    this.selectService(this.state.id);
  }

  selectService = id =>
    this.serviceService.findServiceById(id).then(service => {
      this.props.history.push('/admin/services/' + id);
      this.setState({
        title: service.title,
        id: service.id,
      });
    });

  backToServices = () => {
    this.props.history.push('/admin/services/');
  };

  updateTitle = e => {
    this.setState({ title: e.target.value });
  };

  deleteService = id => {
    this.serviceService.deleteService(id).then(response => {
      if (response.status === 200) {
        this.props.history.push('/admin/services/');
      } else {
        console.log(`Failed to delete service with id ${id}`);
      }
    });
  };

  updateService = () => {
    const { id, title } = this.state;
    let serviceData = { id, title };
    this.serviceService.updateService(id, serviceData).then(response => {
      let newServices = [];
      this.state.services.forEach(service => {
        if (service.id === id) {
          newServices.push(serviceData);
        } else {
          newServices.push(service);
        }
      });
      this.setState({ services: newServices });
      alert('Service updated.');
    });
  };

  render() {
    return (
      <div className="service-detail-container">
        <h3 className="service-detail-header">{this.state.title}</h3>
        <span className="service-detail-row">
          <span className="user-detail-title">Services</span>
          <span className="service-detail-input">
            <select
              value={this.state.id}
              onChange={e => this.selectService(e.target.value)}
              className="form-control"
            >
              {this.state.services.map(service => (
                <option value={service.id} key={service.id}>
                  {service.title}
                </option>
              ))}
            </select>
          </span>
        </span>
        <span className="service-detail-row">
          <span className="service-detail-title">Title</span>
          <span className="service-detail-input">
            <input
              onChange={this.updateTitle}
              className="form-control"
              value={this.state.title}
            />
          </span>
        </span>
        <span className="service-detail-buttons">
          <button
            className="service-detail-cancel"
            onClick={this.backToServices}
          >
            Cancel
          </button>
          <button
            className="service-detail-delete"
            onClick={() => this.deleteService(this.state.id)}
          >
            Delete
          </button>
          <button
            className="service-detail-update"
            onClick={this.updateService}
          >
            Update
          </button>
        </span>
      </div>
    );
  }
}

export default ServiceDetails;
