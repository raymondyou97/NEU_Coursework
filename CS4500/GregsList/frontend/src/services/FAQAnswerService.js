import apiFetch, {
  postJson,
  deleteRequest,
  putRequest,
} from '../util/apiFetch';

export default class FAQAnswerService {
  static instance = null;
  
  static getInstance() {
    if (FAQAnswerService.instance === null) {
      FAQAnswerService.instance = new FAQAnswerService();
    }
    return this.instance;
  }

  findFAQAnswerById = id =>
    apiFetch(`/api/faas/${id}`).then(response => response.json());

  findAllFAQAnswers = () =>
    apiFetch('/api/faas').then(response => response.json());

  createFAQAnswer = (faqId, data) =>
    postJson(`/api/faqs/${faqId}/faas`, data).then(response => response.json());

  updateFAQAnswer = (id, data) =>
    putRequest(`/api/faas/${id}`, data).then(response => response.json());

  deleteFAQAnswer = id => deleteRequest(`/api/faas/${id}`);
}
