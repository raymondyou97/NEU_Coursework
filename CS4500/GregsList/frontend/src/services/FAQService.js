import apiFetch, {
  deleteRequest,
  postJson,
  putRequest,
} from '../util/apiFetch';

export default class FAQService {
  static instance = null;

  static getInstance() {
    if (FAQService.instance === null) {
      FAQService.instance = new FAQService();
    }
    return this.instance;
  }

  findFAQById = id =>
    apiFetch(`/api/faqs/${id}`).then(response => response.json());

  findAllFrequentlyAskedQuestions = (currentPage, perPage, title, question) =>
    apiFetch(
      `/api/faqs?page=${currentPage}&count=${perPage}&titleFilter=${title}&questionFilter=${question}`
    ).then(response => response.json());

  // Data should have a title and question
  createFAQ = data =>
    postJson('/api/faqs', data).then(response => response.json());

  updateFAQ = (id, data) =>
    putRequest(`/api/faqs/${id}`, data).then(response => response.json());

  deleteFAQ = id => deleteRequest(`/api/faqs/${id}`);
}
