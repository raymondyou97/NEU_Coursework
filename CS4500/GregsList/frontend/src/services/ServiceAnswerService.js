import apiFetch, {
  deleteRequest,
  postJson,
  putRequest,
} from '../util/apiFetch';

export default class ServiceAnswerService {
  static instance = null;
  static getInstance() {
    if (ServiceAnswerService.instance === null) {
      ServiceAnswerService.instance = new ServiceAnswerService();
    }
    return this.instance;
  }

  findServiceAnswerById = id =>
    apiFetch(`/api/serviceAnswers/${id}`).then(response => response.json());

  findAllServiceAnswers = () =>
    apiFetch('/api/serviceAnswers').then(response => response.json());

  findPagedServiceAnswers = (page, count) =>
    apiFetch(`/api/serviceAnswers/paged?page=${page}&count=${count}`).then(
      response => response.json()
    );

  createServiceAnswer = data =>
    postJson('/api/serviceAnswers', data).then(response => response.json());

  updateServiceAnswer = (id, data) =>
    putRequest(`/api/serviceAnswers/${id}`, data).then(response =>
      response.json()
    );

  deleteServiceAnswer = id => deleteRequest(`/api/serviceAnswers/${id}`);
}
