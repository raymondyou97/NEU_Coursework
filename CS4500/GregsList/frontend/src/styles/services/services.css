td {
  padding: 16px 8px;
  box-sizing: border-box;
  font-family: 'Times New Roman';
  cursor: pointer;
}

tr {
  padding: 0px 16px;
}

button {
  height: 25px;
  color: white;
  border-radius: 3px;
  border: 0;
}

.service-title {
  margin-bottom: 0;
}

.service-table {
  width: 50%;
  margin: 16px auto;
  display: flex;
  flex: 0 1;
  border: 1px solid #556270;
  border-radius: 3px;
}

.service-table-body {
  display: inline-block;
  width: 100%;
}

.service-details {
  display: flex;
  flex: 1 1;
}

.service-columns {
  border-bottom: 1px solid #333;
  font-weight: bold;
  display: flex;
  flex: 1 1;
}

.service-columns td {
  display: flex;
  flex: 1 0;
  font-size: 20px;
}

.service-details td {
  display: flex;
  box-sizing: border-box;
  flex: 1 0;
}

.service-detail-buttons {
  display: flex;
  flex: 1 1;
}

.service-buttons button {
  width: 30%;
}

.service-buttons button:first-child {
  margin-left: auto;
  margin-right: 4px;
}

.service-inputs {
  display: flex;
  flex: 1 1;
}

.service-inputs td {
  display: flex;
  flex: 1 0;
}

.service-inputs td input {
  width: 100%;
  height: 19px;
  border-radius: 3px;
  border: 1px solid #d5ded9;
}

.service-input-buttons button {
  width: 30%;
}

.service-input-buttons button:first-child {
  margin-left: auto;
  margin-right: 4px;
}

.service-add-button,
.service-edit-button {
  background-color: #4ecdc4;
}

.service-update-button {
  background-color: #ff6b6b;
}

.service-delete-button {
  background-color: #c44d58;
}
