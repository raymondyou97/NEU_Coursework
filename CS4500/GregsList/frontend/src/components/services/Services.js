import React, { Component } from 'react';

import '../../styles/services/services.css';

class Services extends Component {
  componentDidMount() {
    this.props.findAllServices();
  }

  deleteService = id => {
    this.props.deleteService(id);
  };

  enableEdits = (id, title) => {
    this.props.enableEdits(id, title);
  };

  handleTitleChange = e => {
    this.props.handleTitleChange(e.target.value);
  };

  showService = id => {
    this.props.showService(id);
  };

  render() {
    const { newTitle } = this.props;
    return (
      <div>
        <h3>Services</h3>
        <table className="service-table">
          <tbody className="service-table-body">
            <tr className="service-columns">
              <td>Title</td>
            </tr>
            <tr className="service-inputs">
              <td>
                <input
                  className="service-input-title"
                  type="text"
                  placeholder="Title"
                  value={newTitle}
                  onChange={this.handleTitleChange}
                />
              </td>
              <td className="service-input-buttons">
                <button
                  className="service-add-button"
                  onClick={() => this.props.createService()}
                >
                  Add
                </button>
                <button
                  className="service-update-button"
                  onClick={() => this.props.updateService()}
                >
                  Update
                </button>
              </td>
            </tr>
            {this.props.services.map(service => (
              <tr className="service-details" key={service.id}>
                <td onClick={() => this.showService(service.id)}>
                  {service.title}
                </td>
                <td className="service-buttons">
                  <button
                    className="service-delete-button"
                    onClick={() => this.deleteService(service.id)}
                  >
                    Delete
                  </button>
                  <button
                    className="service-edit-button"
                    onClick={() => this.enableEdits(service.id, service.title)}
                  >
                    Edit
                  </button>
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}

export default Services;
