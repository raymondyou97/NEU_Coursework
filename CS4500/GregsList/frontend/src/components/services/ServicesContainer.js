'use es6';

import React, { PureComponent } from 'react';

import ServiceService from '../../services/ServiceService';
import Services from './Services';

class ServicesContainer extends PureComponent {
  constructor(props) {
    super(props);
    this.serviceService = ServiceService.getInstance();
    this.state = {
      services: [],
      newTitle: '',
      selectedId: -1,
    };
  }

  createService = () => {
    this.serviceService
      .createService({
        title: this.state.newTitle,
      })
      .then(respData => {
        this.setState(prevState => ({
          services: prevState.services.concat(respData),
          newTitle: '',
          selectedId: -1,
        }));
      });
  };

  deleteService = id => {
    this.serviceService.deleteService(id).then(response => {
      if (response.status === 200) {
        let newServices = this.state.services.filter(service => {
          return service.id !== id;
        });
        this.setState({ services: newServices });
      } else {
        console.log(`Failed to delete service with id: ${id}`);
      }
    });
  };

  enableEdits = (id, title) => {
    this.setState({ newTitle: title, selectedId: id });
  };

  findAllServices = () => {
    this.serviceService.findAllServices().then(services => {
      this.setState({ services: services });
    });
  };

  handleTitleChange = title => {
    this.setState({ newTitle: title });
  };

  showService = id => {
    this.props.history.push(`/admin/services/${id}`);
  };

  updateService = () => {
    if (this.state.selectedId !== -1) {
      let serviceUpdate = {
        id: this.state.selectedId,
        title: this.state.newTitle,
      };
      this.serviceService
        .updateService(this.state.selectedId, serviceUpdate)
        .then(respData => {
          let newServices = [];
          this.state.services.forEach(service => {
            if (service.id === this.state.selectedId) {
              newServices.push(serviceUpdate);
            } else {
              newServices.push(service);
            }
          });
          this.setState({
            services: newServices,
            newTitle: '',
            selectedId: -1,
          });
        });
    } else {
      console.log('Must have selected a row to update.');
    }
  };

  render() {
    const { newTitle } = this.state;

    return (
      <Services
        newTitle={newTitle}
        findAllServices={this.findAllServices}
        enableEdits={this.enableEdits}
        services={this.state.services}
        handleTitleChange={this.handleTitleChange}
        createService={this.createService}
        showService={this.showService}
        deleteService={this.deleteService}
        updateService={this.updateService}
      />
    );
  }
}

export default ServicesContainer;
