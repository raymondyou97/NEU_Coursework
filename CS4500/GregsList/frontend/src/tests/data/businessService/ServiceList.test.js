import React from 'react';
import renderer from 'react-test-renderer';

import ServiceList from '../../../components/businessService/ServiceList';
import services from './business-service-questions.mock.json';

test('Snapshot renders', () => {
  const component = renderer.create(
    <ServiceList
      services={services}
      selectedService={undefined}
      setSelectedService={() => {}}
      setSelectedServices={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});
