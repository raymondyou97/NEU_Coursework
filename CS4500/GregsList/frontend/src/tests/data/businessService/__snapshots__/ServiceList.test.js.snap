// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders 1`] = `
<div
  className="list-group"
  onKeyDown={[Function]}
>
  <div
    className="list-group-item"
    data-rb-event-key={null}
    disabled={false}
    onClick={[Function]}
  />
</div>
`;
